packages/updatePrescriptionStatus/src/validation/content.ts

```diff
@@ -0,0 +1,166 @@
+  const message = "Prescription ID is invalid."
+  const prescriptionID = task.basedOn?.[0].identifier?.value
+  if (!prescriptionID) {
+    return message
+  }
+  return validatePrescriptionID(prescriptionID) ? undefined : message
+}
+
+export function nhsNumber(task: Task): string | undefined {
+  const message = "NHS number is invalid."
+  const nhsNumber = task.for?.identifier?.value
+  if (!nhsNumber) {
+    return message
+  }
+  return validateNhsNumber(nhsNumber) ? undefined : message
+}
+
+export function resourceType(task: Task): string | undefined {
+  const message = "Resource's resourceType is not 'Task'."
+  const isTask = task.resourceType === "Task"
+  if (!isTask) {
+    return message
+  }
+}
+
+export function codeSystems(task: Task): string | undefined {
+  const systems: Array<Validation> = [
+    (t: Task) => t.focus!.identifier!.system === LINE_ITEM_ID_CODESYSTEM ? undefined : "LineItemID",
+    (t: Task) => t.for!.identifier!.system === NHS_NUMBER_CODESYSTEM ? undefined : "PatientNHSNumber",
+    (t: Task) => t.owner!.identifier!.system === ODS_CODE_CODESYSTEM ? undefined : "PharmacyODSCode",
+    (t: Task) => t.basedOn![0].identifier!.system === PRESCRIPTION_ID_CODESYSTEM ? undefined : "PrescriptionID",
+    (t: Task) => t.businessStatus!.coding![0].system === STATUS_CODESYSTEM ? undefined : "Status"
+  ]
+  const incorrectCodeSystems: Array<string> = []
+  for (const system of systems) {
+    const incorrect = system(task)
+    if (incorrect) {
+      incorrectCodeSystems.push(incorrect)
+    }
+  }
+  if (incorrectCodeSystems.length > 0) {
+    return `Invalid CodeSystem(s) - ${incorrectCodeSystems.join(", ")}.`
+  }
+}
+
+export function businessStatus(task: Task): string | undefined {
+  const code: string = task.businessStatus!.coding![0].code!
+  if (!BUSINESS_STATUSES.includes(code.toLowerCase())) {
+    return "Invalid business status."
+  }
+}
+
+export function statuses(task: Task): string | undefined {
+  const status = task.status
+  if (status === "completed") {
+    const businessStatus: CodeableConcept | undefined = task.businessStatus
+    if (businessStatus) {
+      const coding: Coding = businessStatus.coding![0]
+      const code = coding.code
+      if (code && ["with pharmacy", "ready to collect"].includes(code.toLowerCase())) {
+        return "Completed state indicated for a prescription status requiring patient action."
+      }
+    }
+  }
+}
+
+export function validateContent(task: Task): ValidationOutcome {
+  const contentValidations: Array<Validation> = [
+    businessStatus,
+    lastModified,
+    nhsNumber,
+    prescriptionID,
+    resourceType,
+    statuses,
+    codeSystems
+  ]
+
+  const validationOutcome: ValidationOutcome = {valid: true, issues: undefined}
+
+  const issues: Array<string> = []
+  contentValidations.forEach((validation: Validation) => {
+    try {
+      const issue = validation(task)
+      if (issue) {
+        issues.push(issue)
+      }
+    } catch(e) {
```

Comment:
Thrown exceptions during validation shouldn't result it a 400 unless it's a known thing like a json validation. Best to just not catch here and let it bubble up to the middy handler.

Reply:
Have removed the try/catch.